api/service/explorer/storage.go

```diff
@@ -83,25 +84,40 @@ func (storage *Storage) Dump(block *types.Block, height uint64) {
 	// Store txs
 	for _, tx := range block.Transactions() {
 		explorerTransaction := GetTransaction(tx, block)
-		storage.UpdateAddress(batch, explorerTransaction, tx)
+		storage.UpdateTxAddress(batch, explorerTransaction, tx)
+	}
+	// Store staking txns
+	for _, tx := range block.StakingTransactions() {
+		explorerTransaction := GetStakingTransaction(tx, block)
```

Comment:
as `GetStakingTransaction` could return nil and `UpdateStakingTxAddress` directly dereference it, it is better to check for nil?

Reply:
No. if it is null, that is an unexpected result and we should let the erroneous case go through instead of catching.